web/app/js/components/TopEventTable.jsx

```diff
@@ -16,17 +16,28 @@ const topColumns = (resourceType, ResourceLink, PrefixedLink) => [
   },
   {
     title: "Name",
+    filter: d => {
+      if (d.path === "/ping") {
+        return "";
+      } else if (d.direction === "INBOUND") {
+        return "deploy/" + d.source.owner.split('/')[1];
```

Comment:
this section is a little too application specific (not all applications have a `/ping` endpoint, or have deployments that we're Top-ping).

I think we should filter on whatever the visible name is for this column, e.g. `ns/emojivoto` or `po/web-1234`
To do this, we should return the filter text as whatever is displayed. We'll need a little bit of refactoring so this isn't repetitive:
```diff
diff --git a/web/app/js/components/TopEventTable.jsx b/web/app/js/components/TopEventTable.jsx
index bf4673e7..d98fc3ca 100644
--- a/web/app/js/components/TopEventTable.jsx
+++ b/web/app/js/components/TopEventTable.jsx
@@ -1,10 +1,11 @@
-import { directionColumn, srcDstColumn, tapLink } from './util/TapUtils.jsx';
-import { formatLatencySec, numericSort } from './util/Utils.js';
+import { directionColumn, extractDisplayName, srcDstColumn, tapLink } from './util/TapUtils.jsx';
+import { formatLatencySec, numericSort, toShortResourceName } from './util/Utils.js';

 import BaseTable from './BaseTable.jsx';
 import PropTypes from 'prop-types';
 import React from 'react';
 import SuccessRateMiniChart from './util/SuccessRateMiniChart.jsx';
+import _isEmpty from 'lodash/isEmpty';
 import _isNil from 'lodash/isNil';
 import { withContext } from './util/AppContext.jsx';

@@ -17,13 +18,11 @@ const topColumns = (resourceType, ResourceLink, PrefixedLink) => [
   {
     title: "Name",
     filter: d => {
-      if (d.path === "/ping") {
-        return "";
-      } else if (d.direction === "INBOUND") {
-        return "deploy/" + d.source.owner.split('/')[1];
-      } else if (d.direction === "OUTBOUND") {
-        return "deploy/" + d.destination.owner.split('/')[1];
-      }
+      let [labels, display] = extractDisplayName(d);
+
+      return _isEmpty(labels[resourceType]) ?
+        display.str :
+        toShortResourceName(resourceType) + "/" + labels[resourceType];
     },
     key: "src-dst",
     render: d => srcDstColumn(d, resourceType, ResourceLink)
diff --git a/web/app/js/components/util/TapUtils.jsx b/web/app/js/components/util/TapUtils.jsx
index fe7c9e78..3b95cdec 100644
--- a/web/app/js/components/util/TapUtils.jsx
+++ b/web/app/js/components/util/TapUtils.jsx
@@ -312,8 +312,7 @@ export const directionColumn = d => (
   </Tooltip>
 );

-export const srcDstColumn = (d, resourceType, ResourceLink) => {
-
+export const extractDisplayName = d => {
   let display = {};
   let labels = {};

@@ -324,6 +323,11 @@ export const srcDstColumn = (d, resourceType, ResourceLink) => {
     display = d.destination;
     labels = d.destinationLabels;
   }
+  return [labels, display];
+};
+
+export const srcDstColumn = (d, resourceType, ResourceLink) => {
+  let [labels, display] = extractDisplayName(d);
```

Reply:
Super helpful context and suggestion, thanks! 🌞